src/Rootfs.jl

```diff
@@ -964,3 +964,14 @@ function unmount_shards(ur::Runner; verbose::Bool = false)
     catch
     end
 end
+
+"""
+    installed_shards()
+Return a vector of compiler shards currently downloaded on the local system
+"""
+function installed_shards()
+    hashes = shard_source_artifact_hash.(all_compiler_shards())
+    idx = artifact_exists.(hashes)
```

Comment:
Minor micro-optimisation (fuse all broadcasting and avoid a temporary array):
```suggestion
    idx = artifact_exists.(shard_source_artifact_hash.(all_compiler_shards()))
```